git_wrapper/branch.py

```diff
@@ -80,3 +87,84 @@ def abort_rebase(self):
         except git.GitCommandError as ex:
             msg = "Rebase abort command failed. Error: {0}".format(ex)
             raise_from(exceptions.AbortException(msg), ex)
+
+    @reference_exists('branch_name')
+    def apply_patch(self, branch_name, path):
+        """Apply a git patch file on top of the specified branch.
+
+           :param str branch_name: The name of the branch or reference to apply the patch to
+           :param str path: Path to a git-formatted patch file (cf. git format-patch)
+        """
+        # Expand file (also needed for git-am) and check it exists
+        full_path = self._expand_file_path(path)
+
+        # Checkout
+        try:
+            self.git_repo.git.checkout(branch_name)
+        except git.GitCommandError as ex:
+            msg = "Could not checkout branch {name}. Error: {error}".format(name=branch_name, error=ex)
+            raise_from(exceptions.CheckoutException(msg), ex)
+
+        # Apply the patch file
+        try:
+            self.git_repo.git.am(full_path)
+        except git.GitCommandError as ex:
+            msg = "Could not apply patch {path} on branch {name}. Error: {error}".format(path=full_path, name=branch_name, error=ex)
+            raise_from(exceptions.ChangeNotAppliedException(msg), ex)
+
+    @reference_exists('branch_name')
+    def apply_diff(self, branch_name, diff_path, message, signoff=False):
+        """Apply a diff on top of the specified branch.
+
+           :param str branch_name: The name of the branch or reference to apply the diff to
+           :param str diff_path: Path to the diff file
+           :param str message: Commit message
+           :param bool signoff: Whether to add signed-off-by to commit message
+        """
+        # Ensure we don't commit more than we mean to
+        if self.git_repo.repo.is_dirty():
+            msg = "Repository {repo} contains uncommitted changes. Please clean workspace before proceeding.".format(repo=self.git_repo.repo.working_dir)
```

Comment:
linewrap

Reply:
I would vote for pep8 standard, it is easier to read with you have side by side panes in tmux/vim/whatever

Reply:
I'm somewhat partial to the pep8 standard as well, as I usually have an open file on a half screen and a terminal on the other half.